graphql_compiler/schema_generation/sqlalchemy/edge_descriptors.py

```diff
@@ -56,3 +57,54 @@ def validate_edge_descriptors(vertex_name_to_table, direct_edges):
                 raise InvalidSQLEdgeError('SQL edge {} with edge descriptor {} references a '
                                           'non-existent column {}'
                                           .format(edge_name, direct_edge_descriptor, column_name))
+
+
+def generate_direct_edge_descriptors_from_foreign_keys(vertex_name_to_table):
+    """Return a set of DirectEdgeDescriptor objects from the foreign keys in the SQLAlchemy tables.
+
+    Args:
+        vertex_name_to_table: Dict[str, Table], a mapping of vertex type names to the underlying
+                              SQLAlchemy table objects.
+
+    Return:
+        set of DirectEdgeDescriptor objects extrapolated from the foreign keys.
+    """
+    table_to_vertex_name = _get_table_to_vertex_name(vertex_name_to_table)
+    direct_edge_descriptors = set()
+
+    number_of_composite_foreign_keys = 0
+    for vertex_name, table in vertex_name_to_table.items():
+        for fk_constraint in table.foreign_key_constraints:
+            foreign_key_columns = fk_constraint.columns
+            # The .elements attribute refers to a list of ForeignKey objects.
+            referenced_columns = [element.column for element in fk_constraint.elements]
+            if len(foreign_key_columns) != 1 or len(referenced_columns) != 1:
```

Comment:
This doesn't seem right. If either of these is somehow `0`, we'll still do `number_of_composite_foreign_keys += 1` which is wrong. I think this needs to go from `if-else` to `if-elif-else` with maybe an `AssertionError` in the `else`.